core/lib/analytics/bigcommerce/data-events.ts

```diff
@@ -0,0 +1,83 @@
+import { client } from '~/client';
+import { graphql } from '~/client/graphql';
+import { revalidate } from '~/client/revalidate-target';
+
+const VisitStartedMutation = graphql(`
+  mutation VisitStarted($input: VisitStartedEventInput!) {
+    analytics {
+      visitStartedEvent(input: $input) {
+        executed
+      }
+    }
+  }
+`);
+
+const ProductViewedMutation = graphql(`
+  mutation ProductViewed($input: ProductViewedEventInput!) {
+    analytics {
+      productViewedEvent(input: $input) {
+        executed
+      }
+    }
+  }
+`);
+
+export interface AnalyticsInitiator {
+  visitId: string;
+  visitorId: string;
+}
+
+export interface AnalyticsRequest {
+  url: string;
+  refererUrl: string;
+  userAgent: string;
+}
+
+export interface VisitStartedEvent {
+  initiator: AnalyticsInitiator;
+  request: AnalyticsRequest;
+}
+
+export interface ProductViewedEvent {
+  initiator: AnalyticsInitiator;
+  request: AnalyticsRequest;
+  productId: number;
+}
+
+export async function sendVisitStartedEvent({ initiator, request }: VisitStartedEvent) {
+  const input = { commonInput: preareCommonInput(initiator, request) };
+
+  return client.fetch({
+    document: VisitStartedMutation,
+    variables: { input },
+    fetchOptions: { next: { revalidate } },
+  });
+}
+
+export async function sendProductViewedEvent({
+  productId,
+  initiator,
+  request,
+}: ProductViewedEvent) {
+  const input = {
+    commonInput: preareCommonInput(initiator, request),
+    productInput: { productEntityId: Number(productId) },
+  };
+
+  return await client.fetch({
+    document: ProductViewedMutation,
+    variables: { input },
+    fetchOptions: { next: { revalidate } },
+  });
```

Comment:
We probably want to not cache these requests since they mutations.
```suggestion
  return client.fetch({
    document: VisitStartedMutation,
    variables: { input },
    fetchOptions: { cache: 'no-store' },
  });
}

export async function sendProductViewedEvent({
  productId,
  initiator,
  request,
}: ProductViewedEvent) {
  const input = {
    commonInput: preareCommonInput(initiator, request),
    productInput: { productEntityId: Number(productId) },
  };

  return await client.fetch({
    document: ProductViewedMutation,
    variables: { input },
    fetchOptions: { cache: 'no-store' },
  });
```

Reply:
Nice, I was thinking that was no-cache option, fixed 👍